agent_identity/src/document/aggregate.rs

```diff
@@ -78,139 +97,141 @@ impl Aggregate for Document {
-                                // therefore we need to throw an explixit `InsufficientDepositError` error message.
-                                Err(DIDUpdateError(_, Some(error)))
-                                    if matches!(*error, InputSelection(MissingInputWithEd25519Address)) =>
-                                {
-                                    return Err(InsufficientDepositError(
-                                        network_name.to_string(),
-                                        wallet_address.to_string(),
-                                    ));
-                                }
-                                Err(other_error) => return Err(IotaClientError(other_error)),
+                            // Create a new 'blank' DID Document.
+                            IotaDocument::new(&network_name)
+                        });
+
+                        info!("Testing whether a DID Document can be published...");
+
+                        // This code block is doing a dummy publish to ensure that the DID Document is created and can
+                        // be published/updated later. It uses a gas budget of 0 to avoid actually publishing the
+                        // document.
+                        let document = match identity_client.publish_did_document_update(document.clone(), 0).await {
+                            // This match arm can never be reached, because we use a gas budget of 0.
+                            Ok(document) => document,
+                            // This error occurs when the DID Document is not published yet. We will not return an
+                            // error because the `PublishDocument` command will handle the actual publishing if the
+                            // funds are sufficient.
+                            Err(identity_iota::iota::rebased::Error::DIDResolutionError(_err)) => {
+                                warn!("Document is not published yet.");
+
+                                document
                             }
-                        }
+                            // This error occurs when the `identity_client` has no control over the DID Document that
+                            // has been stored in the aggregate instance. This usually means that the current keys
+                            // stored in the KMS have been updated between boots. We throw an error here indicating
+                            // that the original KMS needs to be used or that the database needs to be wiped.
+                            // TODO: implement KMS migration.
+                            Err(identity_iota::iota::rebased::Error::Identity(err))
+                                if err.contains("address") && err.contains("has no control over Identity") =>
+                            {
+                                warn!("Identity control lost: {err}");
```

Comment:
Can you add a little bit more verbosity here?
`No control over identity since no matching keys could be found in the key storage.` (or similar)